contrib/checkdiff.bash

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MIT
+#
+# Copyright (c) 2021 Rangi
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+FILES=`git diff --name-only $1 HEAD`
```

Comment:
This will break if any of the file names being checked for contains spaces, so...

Solution | Pros | Cons
--- | --- | ---
Don't change anything, instead, document the issue and make sure to never use spaces in our source tree | Easiest to implement, and it's unlikely our source tree will ever contain spaces. | This is prone to subtle breakage later, although somewhat unlikely.
Read the output line by line and store them in an array | Should be reasonably bullet-proof, and doesn't require too large a change. Plus, it avoids re-`grep`ping the entire input on each test. | Does not protect against newlines in file names, but that's pretty unlikely, and this is an "internal" script, so we shouldn't have to worry too much about it.
Pass `-z` to `git-diff` and do the same for each NUL-terminated field | 100% bullet-proof! | I don't know how we'd implement it... and is it worth the trouble? (See above.)

Reply:
The middle option sounds best. The in-PR bash completion script doesn't support newlines either, it's fine.

Reply:
Could add a third argument as an explanation to be printed if the dependency fails, like assertion messages; warning you of what in file 2 goes with file 1.

Reply:
In the end, newline support was only `-z` and `-d ''` (from the last block of https://mywiki.wooledge.org/BashGuide/Arrays#Creating_Arrays-1), so I used the last option, being basically free.